### ISSUE TITLE
Inconsistency in docs?

### ISSUE THREAD

**Issue by pmackay** [vasilakisfil/SimpleAMS#33]
I was trying some more complex nested structures and hit a bug when following the example in the **GraphQL-like APIs** section of https://vasilakisfil.social/blog/2020/01/20/modern-ruby-serializers/. I think where it says `relations: [:microposts]` that should actually read `includes [:microposts]` ? An example further up uses `includes` and when I changed to that it worked as expected.

Is `includes` used in the renderer creation and `relations` inside the DSL? The docs are a _little_ confusing around this point.

**Comment by vasilakisfil** [comment]
I guess you are talking about the DSL.

* `includes` specifies which relations you want to include, out of the available relations.

* `relations` specifies the available relations, so it's not just an array of symbols, but rather full relation objects which are generated through the dsl. The raw representation of relations is an array objects like `[__method__, name, options, embedded_class_for(name, options, block)]`, where `__method__` is the type of the relation (:has_many, :belongs_to etc), name is the name of the relation (like `users`), options, any relation options, and the embedded thing, is options related to that specific relation.

But as I said, you should probably never have to deal with raw relation objects per-se, but rather use the DSL (like `has_many :videos, serializer: VideosSerializer, fields: [:id, :name]`), which will generate the relations for you.

To sum up, yeah the `relations` in the blogpost should be `includes`.

I ll update this weekend the README, with detailed examples, best practices etc. I know I said that like a month ago but things have been hectic lately

**Comment by vasilakisfil** [comment]
Closing. Fixed by #34, it even includes a specific section explaining the difference.

**Comment by pmackay** [comment]
Very helpful reply and much improved docs, many thanks!
